Makefile

```diff
@@ -13,15 +19,23 @@ docker-shell:
 nix-shell:
 	nix-shell
 
-benchmarks:
-	cargo criterion --bench all_benchmarks
+TEST_DIR=proving_system/stark/cairo_programs
+TEST_FILES:=$(wildcard $(TEST_DIR)/*.cairo)
+COMPILED_TESTS:=$(patsubst $(TEST_DIR)/%.cairo, $(TEST_DIR)/%.json, $(TEST_FILES))
+TEST_TRACES:=$(patsubst $(TEST_DIR)/%.cairo, $(TEST_DIR)/%.trace, $(TEST_FILES))
+
+$(TEST_DIR)/%.json: $(TEST_DIR)/%.cairo
+	cairo-compile --cairo_path="$(TEST_DIR)" $< --output $@
```

Comment:
We're missing targets to actually run the program and produce the trace.

Reply:
Yes. I found it easier to produce them inside the benches (as part of the setup, NOT in the benchmarked code). It shouldn't add too much overhead to the total run time. What do you think?

Reply:
In principle, it sounds good. But I'm worried it's not the exact same results. In fact I'm just looking at the `cairo_run` method in the project and it's not using proof mode. It's my understanding that it should run with that enabled.
Ideally I'd use the cairo-lang version for this.

Reply:
Done